### ISSUE TITLE
Add shareable configuration

### ISSUE THREAD

**Issue by remcohaszing** [Gillespie59/eslint-plugin-angular#354]
As pointed out by @AdriVanHoudt in [#258](https://github.com/Gillespie59/eslint-plugin-angular/issues/258#issuecomment-188677966) it might be a nice idea to have a shareable configuration.

New rules added to these defaults, would mean breakage though. Not adding these rules would mean an outdated config.

Upgrading the major version for each new rule seems overkill though.

I suggest to add multiple inheritable configs:

``` yaml
extends:
  - plugin:eslint-plugin-angular/johnpapa
  - plugin:eslint-plugin-angular/bestpractices
  - plugin:eslint-plugin-angular/recommended
```

Where:
- `johnpapa` enables rules related to John Papa's style guide and new rules to enforce John Papa's style guide are not considered breaking changes.
- `bestpractices` enables rules enforcing the AngularJS best practices.
- `recommended` enables some recommended rules and will not change until a new major release is made. E.g. it enables `di-unused`.

I'm not sure how the prefixes required by some rules should be handled.

**Comment by AdriVanHoudt** [comment]
It would only be breaking if you add rules that you would want to enforce in the default but yeah. The multiple config idea sounds good

**Comment by dustinspecker** [comment]
If the maintainers of this plugin are interested, I would be more than glad to add contributors to [eslint-config-angular](https://github.com/dustinspecker/eslint-config-angular) on both GitHub and npm. I haven't been using Angular much recently, so I've definitely been doing a poor job of keeping the config up to date with this plugin.
